frontend/catalyst/pennylane_extensions.py

```diff
@@ -248,22 +321,25 @@ def default_expand_fn(self, circuit, max_expansion=10):
         if any(isinstance(op, MidMeasureMP) for op in circuit.operations):
             raise CompileError("Must use 'measure' from Catalyst instead of PennyLane.")
 
+        decompose_to_qubit_unitary = QJITDevice._get_operations_to_convert_to_matrix(self.config)
+
+        def _decomp_to_unitary(self, *_args, **_kwargs):
+            return [qml.QubitUnitary(qml.matrix(self), wires=self.wires)]
+
         # Fallback for controlled gates that won't decompose successfully.
         # Doing so before rather than after decomposition is generally a trade-off. For low
         # numbers of qubits, a unitary gate might be faster, while for large qubit numbers prior
         # decomposition is generally faster.
         # At the moment, bypassing decomposition for controlled gates will generally have a higher
         # success rate, as complex decomposition paths can fail to trace (c.f. PL #3521, #3522).
-
-        def _decomp_controlled(self, *_args, **_kwargs):
-            return [qml.QubitUnitary(qml.matrix(self), wires=self.wires)]
-
-        with Patcher(
+        overriden_methods = [  # pragma: no cover
             (qml.ops.Controlled, "has_decomposition", lambda self: True),
-            (qml.ops.Controlled, "decomposition", _decomp_controlled),
-            # TODO: Remove once work_wires is no longer needed for decomposition.
-            (qml.ops.MultiControlledX, "decomposition", _decomp_controlled),
-        ):
+            (qml.ops.Controlled, "decomposition", _decomp_to_unitary),
+        ]
```

Comment:
This still puzzles me, as the assignment itself should clearly be covered. Is it the lambda body that is uncovered?

Reply:
Same. I don't think it is the lambda body that is uncovered (but maybe?) deleting either of these two will result in decomposition errors.